bundles/org.openhab.binding.mystrom/src/main/java/org/openhab/binding/mystrom/internal/MyStromBulbHandler.java

```diff
@@ -0,0 +1,314 @@
+        }
+    }
+
+    private void updateDevice(@Nullable MyStromBulbResponse deviceInfo) {
+        if (deviceInfo != null) {
+            updateState(CHANNEL_SWITCH, deviceInfo.on ? OnOffType.ON : OnOffType.OFF);
+            @Nullable
+            HSBType color = null;
+            long numSemicolon = deviceInfo.color.chars().filter(c -> c == ';').count();
+            if (numSemicolon == 1 && deviceInfo.mode.equals(MONO)) {
+                String[] xy = deviceInfo.color.split(";");
+                lastColorTemperature = new QuantityType<>(xy[0]);
+                lastBrightness = PercentType.valueOf(xy[1]);
+                updateState(CHANNEL_COLOR_TEMPERATURE, lastColorTemperature);
+                updateState(CHANNEL_BRIGHTNESS, lastBrightness);
+                updateState(CHANNEL_MODE, StringType.valueOf(deviceInfo.mode));
+            } else if (numSemicolon == 2 && deviceInfo.mode.equals(HSV)) {
+                color = HSBType.valueOf(deviceInfo.color.replaceAll(";", ","));
+            } else if (!deviceInfo.color.equals("") && deviceInfo.mode.equals(RGB)) {
+                int r = Integer.parseInt(deviceInfo.color.substring(2, 4), 16);
+                int g = Integer.parseInt(deviceInfo.color.substring(4, 6), 16);
+                int b = Integer.parseInt(deviceInfo.color.substring(6, 8), 16);
+                color = HSBType.fromRGB(r, g, b);
+            }
+            if (color != null) {
+                updateState(CHANNEL_COLOR, color);
+                updateState(CHANNEL_MODE, StringType.valueOf(deviceInfo.mode));
+            }
+            updateState(CHANNEL_RAMP, QuantityType.valueOf(deviceInfo.ramp, MetricPrefix.MILLI(SECOND)));
+            if (deviceInfo instanceof MyStromDeviceSpecificInfo) {
+                updateState(CHANNEL_POWER, QuantityType.valueOf(((MyStromDeviceSpecificInfo) deviceInfo).power, WATT));
+            }
+        }
+    }
+
+    /**
+     * Given a URL, send a HTTP GET request to the URL location
+     * created by the URL.
+     *
+     * @return String contents of the response for the GET request.
+     * @throws MyStromException Throws on communication error
+     */
+    private String getDeviceSpecificInformation() throws MyStromException {
+        String url = getHostname() + "/api/v1/device";
+        ContentResponse response;
+        try {
+            Request request = getHttpClient().newRequest(url).timeout(10, TimeUnit.SECONDS).method(HttpMethod.GET);
+            response = request.send();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new MyStromException(COMMUNICATION_ERROR + e.getMessage());
+        }
+
+        if (response.getStatus() != HTTP_OK_CODE) {
+            throw new MyStromException(
+                    "Error sending HTTP GET request to " + url + ". Got response code: " + response.getStatus());
+        }
+        return response.getContentAsString();
+    }
```

Comment:
```suggestion
    private String getDeviceSpecificInformation() throws MyStromException {
        String url = getHostname() + "/api/v1/device";
        try {
            Request request = getHttpClient().newRequest(url).timeout(10, TimeUnit.SECONDS).method(HttpMethod.GET);
            ContentResponse response = request.send();
            if (response.getStatus() != HTTP_OK_CODE) {
                throw new MyStromException(
                    "Error sending HTTP GET request to " + url + ". Got response code: " + response.getStatus());
            }
            return response.getContentAsString();   
        } catch (InterruptedException | TimeoutException | ExecutionException e) {
            throw new MyStromException(COMMUNICATION_ERROR + e.getMessage());
        }
    }
```

Also in other occurences. Usually it makes code more readable to keep it to gether and not separate it with exception handling code.